### ISSUE TITLE
Add support for non-subresource request parameters [JIRA: RCS-167]

### ISSUE THREAD

**Issue by kellymclaughlin** [basho/riak_cs#582]
Support the `response` request parameters described in the `Request Parameters` section of [this](http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectGET.html) page. _e.g._ `response-content-type` to indicate a `Content-Type` that should be used for the response.

The lack of this support was uncovered by running the `test_object_raw_response_headers` test in the [`s3-tests`](https://github.com/basho/s3-tests) suite. It has been marked as a skipped tests, but once this change is finished it should be verified and included back in the testing suite.

**Comment by shino** [comment]
Duplicate of #736
